python_bindings/generate.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+import os
+import argparse
+from collections import OrderedDict
+
+try:
+    # For generating bindings
+    import cffi 
+except (ImportError):
+    print("[!] `CFFI` is missing.")
+    exit(1)
+
+# pycparser is a dependency of cffi, so must be available here
+from pycparser import parse_file, c_generator, c_ast 
+
+# The C pre-processor expands the fake stdlib headers we provide. If this is compiled by cffi, there will be weird type size issues.
+def kill_unwanted_typedefs(tree):
+    for x, i in tree.children():
+        # We only need to look at Typedefs
+        if type(i) is c_ast.Typedef:
+            # If the type is too simple or if it's a struct with no implementation
+            if type(i.type.type) is c_ast.IdentifierType or (type(i.type.type) is c_ast.Struct and not i.type.type.decls):
+                tree.ext.remove(i)
+            # Some edge cases - a typedef with pointers
+            elif type(i.type) is c_ast.PtrDecl:
+                if type(i.type.type.type) is c_ast.IdentifierType:
+                    tree.ext.remove(i)
+
+
+def expose_decl_in_bindings(files: list, ffi: cffi.FFI, c_compiler: str):
+    # Create a temporary file as workaround for issues with redeclaring types
+    with open('.tmp_preprocess.h', 'w') as tmp_file:
+        tmp_file.write(format_include_for_compiler(files))
```

Comment:
https://docs.python.org/3/library/io.html

Reply:
Memory files

Reply:
I needed an actual file on disk, so I instead used the python tempfile module. Is that satisfactory?

Reply:
Yeah sure.